cpp/models/ode_seir/model.h

```diff
@@ -66,38 +67,35 @@ class Model : public CompartmentalModel<InfectionState, Populations<InfectionSta
     }
 
     /**
-    *@brief After the simulation is finished and we get a resulting TimeSeries, this function uses the data to compute the reproduction number
-    *at an arbitrary time for this specific SEIR model.
-    *@param t_idx The time at which we want to compute the reproduction number.
-    *@param y The TimeSeries. We actually only use the number of Susceptibles at a given time.
-    *@returns The reproduction number in the seir model.
-    *@see The same functions in the model.h files of the secir and secirvvs models.
+    *@brief Computes the reproduction number at a given index time of the Model output obtained by the Simulation.
+    *@param t_idx The index time at which the reproduction number is computed.
+    *@param y The TimeSeries obtained from the Model Simulation.
+    *@returns The computed reproduction number at the provided index time.
     */
-
-    IOResult<double> get_reproduction_number(double t_idx, const mio::TimeSeries<ScalarType>& y) noexcept
-    { 
-        if(!( 0 <= t_idx && t_idx < y.get_num_time_points())){
+    IOResult<double> get_reproduction_number(size_t t_idx, const mio::TimeSeries<ScalarType>& y) noexcept
```

Comment:
why `noexcept`?

Reply:
I added this since they always use it in the documentation of boost::outcome::result<>. The functions that are called are not declared noexcept. But they only accessed parameters and values. Should I remove it ?

Reply:
Do you have a qualitative answer on that @dabele ?

Reply:
I guess the boost::coutcome documentation does that to highlight that they are not using exceptions for error handling, they are using return values, which is exactly what boost::outcome is for. 

Since we are also not using exceptions for error handling, we could put noexcept almost everywhere. Some people would consider this more correct. Some minor optimizations in the STL and compiler are enabled by this.

I don't see any real benefit or harm either way.